tools/syz-headerparser/headerparser.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2015 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+'''
+This scripts takes as input a list of header files, a device name and generates metadata
+files to make syzkaller device descriptions.
+'''
+
+import argparse
+import logging
+
+from headerlib.container import GlobalHierarchy
+
+
+def main():
+    """
+    python parser.py --device=test --filename=A.h,B.h
+    """
+
+    parser = argparse.ArgumentParser(description='Parse header files to output fuzzer'
+                                                 'struct metadata.')
+    parser.add_argument('--device',
+                        help='name of the device you want to write a fuzzer for',
+                        dest='device',
+                        required=True)
+    parser.add_argument('--filenames',
+                        help='comma-separated header filenames',
+                        dest='filenames',
+                        required=True)
+    parser.add_argument('--debug',
+                        help='print debug-information at every level of parsing',
+                        action='store_true')
+    parser.add_argument('--template',
+                        help='add lines from <template> into the template in headerpreprocessor',
+                        required=False,
+                        const='',
+                        nargs='?')
+    parser.add_argument('--outputfmt',
+                        help='the fuzzer for which the output will be formatted',
+                        required=False,
+                        default='syzkaller',
+                        nargs='?')
+
+    args = parser.parse_args()
+
+    loglvl = logging.INFO
+
+    if args.debug:
+        loglvl = logging.DEBUG
+
+    template_lines = ''
+    if args.template:
+        template_lines = open(args.template, 'r').read()
+
+    gh = GlobalHierarchy(device_name=args.device, filenames=args.filenames.split(','),
+                         loglvl=loglvl, template_lines=template_lines,
+                         output_fmt=args.outputfmt)
+
+    print '-=============== Structures -]'
```

Comment:
What is Structures? Why is it surrounded with this strange punctuation?
Do we need `print str(gh)` at all? It is empty in both provided examples. What is it supposed to contain?
I would replace these 4 lines with just:
```
print gh.get_metadata_structs()
```